python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -0,0 +1,1753 @@
+            if getattr(self.check, cmp):
+                self.log.debug(f"Adding {cmp} to check list")
+                coro_list.append(asyncio.ensure_future(self.check_component_state(cmp)))
+                # TODO: Implement verify method
+                # coro_list.append(asyncio.ensure_future(self.verify(cmp)))
+                # TODO: Not all components publish heartbeats!
+                # coro_list.append(asyncio.ensure_future(self.check_component_hb(cmp)))
+            else:
+                self.log.debug(f"Skipping {cmp}")
+
+        for res in asyncio.as_completed(coro_list):
+            try:
+                await res
+            except Exception:
+                raise
+            else:
+                break
+            finally:
+                await self.cancel_not_done(coro_list)
+
+    async def startup(self, settings=None):
+        """Startup ATTCS components.
+
+        This method will perform the start of the night procedure for the
+        ATTCS component. It will enable all components, open the dome slit,
+        and open the telescope covers.
+
+        Parameters
+        ----------
+        settings: `dict`
+            Dictionary with settings to apply.  If `None` use the recommended
+            settings.
+        """
+
+        await self.enable(settings=settings)
+
+        self.log.debug("Slew telescope to park position.")
+        self.check.atdometrajectory = False
+        atdome_check = self.check.atdome
+        await self.point_azel(
+            target_name="Park position",
+            az=self.tel_park_az,
+            el=self.tel_park_el,
+            wait_dome=False,
+        )
+        self.check.atdome = atdome_check
+
+        try:
+            await self.stop_tracking()
+        except asyncio.TimeoutError:
+            pass
+
+        # Close m1 cover if it is open.
+        await self.close_m1_cover()
+
+        if self.check.atdome:
+
+            self.log.debug("Homing dome azimuth.")
+
+            await self.home_dome()
+
+            self.log.debug("Moving dome to 90 degrees.")
+            await self.slew_dome_to(az=90.0)
+
+            self.log.info(
+                "Check that dome CSC can communicate with shutter control box."
+            )
+
+            try:
+                scb = await self.atdome.evt_scbLink.aget(timeout=self.fast_timeout)
+            except asyncio.TimeoutError:
+                self.log.error(
+                    "Timed out waiting for ATDome CSC scbLink status event. Can not "
+                    "determine if CSC has communication with Shutter Control Box."
+                    "If running this on a jupyter notebook you may try to add an"
+                    "await asyncio.sleep(1.) before calling startup again to give the"
+                    "remotes time to get information from DDS. You may also try to "
+                    "re-cycle the ATDome CSC state to STANDBY and back to ENABLE."
+                    "Cannot continue."
+                )
```

Comment:
I don't know if we actually want that to happen.. we need to talk to @patrickingraham  and see what he thinks but we actually don't need the link to be up to do some operations (like setting up for flat field) and, in fact, if the dome is open we can observe with the link down. 

I think we may want to eventually add a rule in the watcher to look for this but I don't think we want to prevent operation because the link is down.

Reply:
So this is a bug that the vendor is supposed to provide a fix for. They've actually provided code to help diagnose the issue but because of the closure we obviously can't test it. Once this behaviour is fixed then going to FAULT is appropriate, but for the moment if we implement it then we'll just end up losing sky time. Does that help?

Reply:
Thanks. Let's just leave the ATDome CSC alone then.

If the low level controller only tried to connect to the SCB when the CSC connected to the low level controller then I'd suggest changing the CSC to wait a few seconds for that to happen.

But I hope and assume that the low level controller normally tries to maintain a connect to the SCB whenever the system is on.